### PR TITLE
[SM68] Add feature flag for SampleCmpBias/Grad and StartInstance/VertexLocation

### PR DESCRIPTION
**Pull request by python3kgae**
New feature flag ShaderFeatureInfo_SampleCmpGradientOrBias = 0x80000000 and 
ShaderFeatureInfo_ExtendedCommandInfo = 0x100000000 were added. 
 They will be set when SampleCmpBias/Grad startInstance/VertexLocation exists.

### CODE REVIEW

File: include/dxc/DXIL/DxilConstants.h [hunk 1]
```diff
@@ -1888,7 +1888,9 @@ const uint64_t ShaderFeatureInfo_WriteableMSAATextures = 0x40000000;
 // WaveMMA slots in between two SM 6.6 feature bits.
 const uint64_t ShaderFeatureInfo_WaveMMA = 0x8000000;
 
-const unsigned ShaderFeatureInfoCount = 31;
+const uint64_t ShaderFeatureInfo_SampleCmpGradientOrBias = 0x80000000;
```
Comment (tex3d) [inline include/dxc/DXIL/DxilConstants.h:1891]: Need to add both flags at once, see: #6146.

File: tools/clang/tools/dxcompiler/dxcdisassembler.cpp [hunk 2]
```diff
@@ -347,6 +347,7 @@ PCSTR g_pFeatureInfoNames[] = {
     "64-bit Atomics on Heap Resources",
     "Advanced Texture Ops",
     "Writeable MSAA Textures",
+    "SampleCmpGradientOrBias",
```
Comment (tex3d) [inline tools/clang/tools/dxcompiler/dxcdisassembler.cpp:350]: Need to add both flags at once, see: #6146.

File: include/dxc/DXIL/DxilConstants.h (hunk 1 above)
Comment (python3kgae) [reply include/dxc/DXIL/DxilConstants.h:1891]: Added.

File: tools/clang/tools/dxcompiler/dxcdisassembler.cpp (hunk 2 above)
Comment (python3kgae) [reply tools/clang/tools/dxcompiler/dxcdisassembler.cpp:350]: Added.

Comment (tex3d) [reply tools/clang/tools/dxcompiler/dxcdisassembler.cpp:350]: I think we should be using the description strings from #6146:
```
    "SampleCmp with gradient or bias",
    "Extended command info",
```

File: tools/clang/test/HLSLFileCheck/hlsl/objects/Texture/has_lod_clmap.hlsl [hunk 3]
```diff
@@ -2,22 +2,31 @@
 // RUN: %dxilver 1.8 | %dxc -E test_sampleb -T ps_6_8 %s | FileCheck %s
 // RUN: %dxilver 1.8 | %dxc -E test_sampleg -T ps_6_8 %s | FileCheck %s
 // RUN: %dxilver 1.8 | %dxc -E test_samplec -T ps_6_8 %s | FileCheck %s
-// RUN: %dxilver 1.8 | %dxc -E test_samplecb -T ps_6_8 %s | FileCheck %s
-// RUN: %dxilver 1.8 | %dxc -E test_samplecg -T ps_6_8 %s | FileCheck %s
+// RUN: %dxilver 1.8 | %dxc -E test_samplecb -T ps_6_8 %s | FileCheck %s -check-prefix=CMPBG
+// RUN: %dxilver 1.8 | %dxc -E test_samplecg -T ps_6_8 %s | FileCheck %s -check-prefix=CMPBG
```
Comment (pow2clk) [inline tools/clang/test/HLSLFileCheck/hlsl/objects/Texture/has_lod_clmap.hlsl:6]: You can have multiple prefixes in the same run if you use -check-prefixes=CHECK,CMPBG, which would avoid duplicated lines below. Not crucial, but nice.

Comment (python3kgae) [reply tools/clang/test/HLSLFileCheck/hlsl/objects/Texture/has_lod_clmap.hlsl:6]: Not work for taef.

Comment (python3kgae) [reply tools/clang/test/HLSLFileCheck/hlsl/objects/Texture/has_lod_clmap.hlsl]: Fixed.

File: tools/clang/tools/dxcompiler/dxcdisassembler.cpp (hunk 2 above)
Comment (pow2clk) [reply tools/clang/tools/dxcompiler/dxcdisassembler.cpp:350]:
```suggestion
    "SampleCmp with gradient or bias",
```

File: tools/clang/tools/dxcompiler/dxcdisassembler.cpp [hunk 4]
```diff
@@ -347,6 +347,8 @@ PCSTR g_pFeatureInfoNames[] = {
...
+    "ExtendedCommandInfo",
```
Comment (pow2clk) [inline tools/clang/tools/dxcompiler/dxcdisassembler.cpp:351]:
```suggestion
    "Extended command info",
```

**Review by pow2clk — COMMENTED** [review]
Just trying to facilitate Tex's suggestions

File: tools/clang/tools/dxcompiler/dxcdisassembler.cpp (hunk 2 above)
Comment (tex3d) [reply tools/clang/tools/dxcompiler/dxcdisassembler.cpp:350]: Note there are references to these in tests that will have to be fixed up as well, so it's not simply a matter of accepting this suggestion.

Comment (python3kgae) [reply tools/clang/tools/dxcompiler/dxcdisassembler.cpp]: Fixed.

File: tools/clang/tools/dxcompiler/dxcdisassembler.cpp (hunk 4 above)
Comment (python3kgae) [reply tools/clang/tools/dxcompiler/dxcdisassembler.cpp]: Fixed.

File: include/dxc/DXIL/DxilConstants.h [hunk 5]
```diff
@@ -1888,7 +1888,10 @@ const uint64_t ShaderFeatureInfo_WriteableMSAATextures = 0x40000000;
...
+const uint64_t ShaderFeatureInfo_ExtendedCommandInfo = 0x100000000;
```
Comment (tex3d) [inline include/dxc/DXIL/DxilConstants.h:1892]: For the record, I don't think the name of this feature bit is intuitive at all.

File: lib/DXIL/DxilShaderFlags.cpp [hunk 6]
```diff
@@ -761,10 +780,11 @@ ShaderFlags ShaderFlags::CollectShaderFlags(const Function *F,
   flag.SetAdvancedTextureOps(hasAdvancedTextureOps);
   flag.SetWriteableMSAATextures(hasWriteableMSAATextures);
   flag.SetWaveMMA(hasWaveMMA);
-
+  flag.SetSampleCmpGradientOrBias(hasSampleCmpGradientOrBias);
```
Comment (tex3d) [inline lib/DXIL/DxilShaderFlags.cpp:783]: nit: Strange ordering, why not add the two lines together at the end?

**Review by tex3d — APPROVED** [review]
LGTM!

**Review by tex3d — APPROVED** [review]
Looks good.  (meant to approve last time)

**Review by pow2clk — APPROVED** [review]
Looks good! Thanks for the quick response!

File: include/dxc/DXIL/DxilConstants.h (hunk 5 above)
Comment (pow2clk) [reply include/dxc/DXIL/DxilConstants.h:1892]: Agreed. It's a remnant from what it was before.

File: lib/DXIL/DxilShaderFlags.cpp (hunk 6 above)
Comment (python3kgae) [reply lib/DXIL/DxilShaderFlags.cpp]: Fixed.

**Pull request merged**